### PR TITLE
Feature/initial code cb 6812

### PR DESCRIPTION
**Pull request by CombeeMike**
[Paymo Task](https://app.paymoapp.com/#Paymo.Projects/2013095/tasks/simple/task/22169148)

Shared TS config which is published to new NPM package [@combeenation/ts-config](https://www.npmjs.com/package/@combeenation/ts-config).
The config is based on the existing ones from CJS utils & the rebuilt project.
I tried to omit the settings which are too project specific like support for JSX files which is only required in react etc.

Usage see other PRs.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
+    "target": "ES6", // Specify ECMAScript target version
+    "lib": ["ES2017", "DOM"], // List of library files to be included in the compilation
+    "allowJs": true, // Allow JavaScript files to be compiled
+    "skipLibCheck": true, // Skip type checking of all declaration files
+    "esModuleInterop": true, // Disables namespace imports (import * as fs from "fs") and enables CJS/AMD/UMD style imports (import fs from "fs")
```
Comment (HoferMarkus) [inline tsconfig.json]: I think that comment is a bit misleading.
As far as I understood the [official docs](https://www.typescriptlang.org/tsconfig#esModuleInterop) a namespace import (`import * as fs from "fs"`) is still valid, but it just gets compiled in a different way.

**Review by HoferMarkus — APPROVED** [review]
LGTM

File: tsconfig.json (hunk 1 above)
Comment (CombeeMike) [reply tsconfig.json]: Ok, to be honest, I think I just copy pasted all those comments from somewhere at some point and did not really question them 🙈

What would you suggest?
Is the following ok?

> Fixes certain issues with namespace imports (import * as fs from "fs") in compiled code

Comment (HoferMarkus) [reply tsconfig.json]: ok, well that suggestion is fine for me

**Review by cbn-falias — APPROVED** [review]
LGTM

**Pull request merged**